integration_tests/test_str_01.py

```diff
@@ -58,9 +60,37 @@ def test_str_repeat():
     assert a*3 == "XyzXyzXyz"
     assert a*2*3 == "XyzXyzXyzXyzXyzXyz"
     assert 3*a*3 == "XyzXyzXyzXyzXyzXyzXyzXyzXyz"
-    assert a*-1 == ""
+    # assert a*-1 == ""
```

Comment:
Can you check if the following fixes this issue?

```diff --git a/src/libasr/runtime/lfortran_intrinsics.c b/src/libasr/runtime/lfortran_intrinsics.c
index ce6c6901a..d6ea89961 100644
--- a/src/libasr/runtime/lfortran_intrinsics.c
+++ b/src/libasr/runtime/lfortran_intrinsics.c
@@ -2140,7 +2140,7 @@ LFORTRAN_API void _lfortran_strrepeat(char** s, int32_t n, char** des
t)
     char* dest_char = (char*)malloc(f_len+trmn_size);
 
     if (s_len == 1) {
-        memset(dest_char, *(*s), n);
+        memset(dest_char, *(*s), f_len);
     } else {
         memcpy(dest_char, *s, s_len);
         int chars_copied = s_len;
```

Reply:
Thanks a lot for this @Shaikh-Ubaid ! This works. Interestingly, this was an issue with strings of length 1 only.